File: backend/app/schemas/refund.py
from datetime import datetime
from typing import Optional, Any
from decimal import Decimal
from pydantic import BaseModel


class RefundCreate(BaseModel):
    booking_id: int
    reason: Optional[str] = None


class RefundRead(BaseModel):
    refund_id: int
    booking_id: int
    gateway_refund_id: Optional[str] = None
    amount: Decimal
    status: str
    reason: Optional[str] = None
    is_manual: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/refund.ts
export type RefundStatus = "PENDING" | "PROCESSING" | "SUCCESS" | "FAILED" | string;

export type Refund = {
  refund_id: number;
  booking_id: number;
  gateway_refund_id?: string | null;
  amount: string | number;
  status: RefundStatus;
  reason?: string | null;
  is_manual: boolean;
  created_at: string;
  updated_at: string;
};

